src/Type/Symfony/ParameterDynamicReturnTypeExtension.php

```diff
@@ -139,16 +141,30 @@ private function getGetTypeFromMethodCall(
 	private function generalizeTypeFromValue(Scope $scope, $value): Type
 	{
 		if (is_array($value) && $value !== []) {
-			return $this->generalizeType(
-				new ArrayType(
-					TypeCombinator::union(...array_map(function ($item) use ($scope): Type {
-						return $this->generalizeTypeFromValue($scope, $item);
-					}, array_keys($value))),
-					TypeCombinator::union(...array_map(function ($item) use ($scope): Type {
-						return $this->generalizeTypeFromValue($scope, $item);
-					}, array_values($value)))
-				)
-			);
+			if (array_is_list($value)) {
```

Comment:
I'd say not to use these functions, although it might work because PHPStan includes these polyfills as well.

One thing you need to handle here is yes/no/maybe:

1) yes - array is list, array should be generalized
2) no - array is indexed, do the logic as you described
3) keys are mixed, some are integers, some are strings - array should also be generalized

Please also add a test for the scenario 3).

Reply:
> I'd say not to use these functions, although it might work because PHPStan includes these polyfills as well.

So how do you prefer I test if it's a list ?

Reply:
Iterate over the keys, see if all are integers or if all are strings, or if its a mix of strings/integers.

Reply:
Sure, it's done :)